src/main/java/java_dz_4/DoctorController.java

```diff
@@ -0,0 +1,77 @@
+package java_dz_4;
+
+
+import lombok.AllArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@AllArgsConstructor
+public class DoctorController {
+    private final DoctorService doctorService;
+
+    @GetMapping("/doctors/{id}")
+    public Doctor findById(@PathVariable Integer id) {
+        Optional<Doctor> mayBeDoctor = doctorService.findById(id);
+        return mayBeDoctor.orElseThrow(DoctorNotFoundException::new);
+    }
+
+    @GetMapping("/doctors")
+    public List<Doctor> findAll() {
+        return doctorService.findAll();
+    }
+
+    @GetMapping("/doctors/specialization={specialization}")
+    public List<Doctor> findDoctorsBySpecialization(@PathVariable String specialization) {
+        return doctorService.findDoctorsBySpecialization(specialization);
+    }
+
+
+    @GetMapping("/doctors/name={letter}")
+    public List<Doctor> findDoctorsByFirstLetter(@PathVariable char letter) {
```

Comment:
That 'letter' should be a request rarameter (GET /doctors?name=Alex) rather than path variable.